### ISSUE TITLE
RMSD and PBCs

### ISSUE THREAD

**Issue by GiovanniBussi** [plumed/plumed2#331]
I think we still have some inconsistency in the treatment of PBCs (see #128).

I would like to add implicit treatment of PBCs to the following cases:
- `RMSD`
- `DIPOLE`
- `FIT_TO_TEMPLATE`
- `PATHMSD`
- `PROPERTYMAP`

(the last two share the same implementation). Am I missing some?

In particular, I will make it such that PBCs are reconstructed by default with `makeWhole()` following atom order (notice that atom order from PDB is preserved, so this should allow using also out-of-order atom lists). It will break backward compatibility, but I think will simplify the input allowing people to avoid the `WHOLEMOLECULES` action in many cases.  I will allow the flag `NOPBC` to go back to present behavior, for consistency with other actions.

I think I should be able to do without interfering with the `reference` data structure, but I might ask @gtribello to double check before merging.

**Comment by carlocamilloni** [comment]
@giovannibussi I think there are still cases that can be problematic like secondary structures cvs using type optimal

**Comment by GiovanniBussi** [comment]
@carlocamilloni you are right.  So, pending actions are:
- `PCAVARS` and `PCARMSD` (done on branch fix-331b)
- Secondary structure variables (todo)

**Comment by GiovanniBussi** [comment]
@carlocamilloni @gtribello I merged `PCAVARS` and `PCARMSD`.

I am not sure how to proceed about the following missing cases:
- Secondary structure variables. In this case, I cannot use the `makeWhole()` call at the beginning of the `calculate()` method. Indeed, this would make the whole list of requested atoms whole, which is not what we want. I think we should make whole every single motif. In order to do this I have to copy the code of `ActionAtomistic::makeWhole()`. (as a side note, I would rather make a separate `Tools/MakeWhole` and then use it in multiple parts of the code - inside `ActionAtomistic::makeWhole()`, inside `WHOLEMOLECULES` and in secondary structure stuff). However, I don't know where to plug the change in secondary structure calculation. Any ides?

- `MULTI-RMSD`. Here I noticed that there is already a `NOPBC` flag that can be used in the pdb file. Should we treat it consistently? Maybe @gtribello has some idea

Giovanni
